app/scripts/controllers/metametrics.ts

```diff
@@ -470,7 +498,22 @@ export default class MetaMetricsController {
     const fragment = fragments[id];
 
     if (!fragment) {
-      throw new Error(`Event fragment with id ${id} does not exist.`);
+      if (!id.includes('transaction-submitted-')) {
```

Comment:
Similar to the other comment, could we avoid the coupling with an optional property in the fragment such as `createIfNotFound`?

Reply:
updated https://github.com/MetaMask/metamask-extension/pull/28240/commits/54d3cb36c1a395400447bcd6341bb7823846829e